Sources/Purchasing/ConfigurationInCustomEntitlementsComputation.swift

```diff
@@ -0,0 +1,229 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  Configuration.swift
+//
+//  Created by Joshua Liebowitz on 5/6/22.
+
+import Foundation
+
+#if ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION
+
+// swiftlint:disable type_name
+/**
+ * ``ConfigurationInCustomEntitlementsComputation`` can be used when configuring the ``Purchases`` instance.
+ * It is not required to be used, but highly recommended. This class follows a builder pattern.
+ *
+ * To configure your `Purchases` instance using this object, follow these steps.
+ *
+ * **Steps:**
+ * 1. Call ``ConfigurationInCustomEntitlementsComputation/builder(withAPIKey:)``
+ *  To obtain a ``ConfigurationInCustomEntitlementsComputation/Builder`` object.
+ * 2. Set this builder's properties using the "`with(`" functions.
+ * 3. Call ``ConfigurationInCustomEntitlementsComputation/Builder/build()`` to obtain the
+ *  `ConfigurationInCustomEntitlementsComputation` object.
+ * 4. Pass the `ConfigurationInCustomEntitlementsComputation` object into
+ *  ``Purchases/configureInCustomEntitlementsComputationMode(with:)``.
+ *
+ * ```swift
+ * let configuration = ConfigurationInCustomEntitlementsComputation.Builder(withAPIKey: "MyKey",
+ *                                                                          appUserID: "SomeAppUserID).build()
+ *  Purchases.configure(with: configuration)
+ * ```
+ */
+public final class ConfigurationInCustomEntitlementsComputation: NSObject {
```

Comment:
Do we need the `@objc` attribute here as in `Configuration`? If not, then does it need to be `NSObject` subclass?

Reply:
Yeah I also had doubts about that... I don't think it's needed TBH, since this shouldn't be used by any hybrids and customers of this mode are using Swift... So yeah, I will remove NSObject here. Good catch!

Reply:
Does not apply anymore after removing this class